### ISSUE TITLE
Error upgrading to v 4.17

### ISSUE THREAD

**Issue by nbabel** [Esri/angular-cli-esri-map#86]
I have a project based on this repo.  It has been working fine with v 4.16.  I wanted to upgrade to v 4.17.  I changed the package.json to 4.17 for the webpack-plugin and the types.  I also updated the web workers per this page:  https://github.com/Esri/arcgis-webpack-plugin#usage

When I run npm install I get a strange shell window that says templates are not found in ...AppData\Local\Temp\pacote-git-template\git-clone-d24dlb2d although the install completes.  When I run ng serve the application won't start.  I get a TypeError: 

TypeError [ERR_INVALID_ARG_TYPE]: The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received undefined    at Function.from (buffer.js:313:9)

When I switch back to 4.16 and do npm install and ng serve it works again as expected.  Is v 4.17 ready to use with webpack and angular?

**Comment by andygup** [comment]
I haven't seen that error before. What's the pathname for `buffer.js`? Is that a JS API module?

I'll run some tests on my end.

**Comment by andygup** [comment]
@nbabel I didn't see the same issues you reported but I also wasn't able to run the webpack sample using 4.17. I'll need to investigate further. Here's a screenshot of the errors I'm seeing:

![image](https://user-images.githubusercontent.com/510440/96617505-f34f1880-12c0-11eb-9234-9778d4d51fac.png)

**Comment by andygup** [comment]
Okay, the problem I'm seeing is when you upgrade the webpack plugin you also need to set the `DEFAULT_WORKER_URL` to the same base version of the plugin. So once I did the following then the app worked for me:

```js

const DEFAULT_WORKER_URL = "https://js.arcgis.com/4.17/";

```

**Comment by nbabel** [comment]
Thanks @andygup, I have set the DEFAULT_WORKER_URL to the same as the base version of the plugin, but it's still not working.
Here is the full warning I get in a shell window when I run npm install:

`warning: templates not found in C:\Users\NBabel\AppData\Local\Temp\pacote-git-template-tmp\git-clone-0c79d1e2`

Here is the full error I'm getting from ng serve:

```
95% emitting index-html-webpack-pluginbuffer.js:313
  throw new ERR_INVALID_ARG_TYPE(
  ^

TypeError [ERR_INVALID_ARG_TYPE]: The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received undefined
    at Function.from (buffer.js:313:9)
    at writeOut (C:\Projects\WFDSS\NextGen\repo\code\UI_NextGen\node_modules\webpack\lib\Compiler.js:413:27)
    at Immediate._onImmediate (C:\Projects\WFDSS\NextGen\repo\code\UI_NextGen\node_modules\memory-fs\lib\MemoryFileSystem.js:288:4)
    at processImmediate (internal/timers.js:456:21) {
  code: 'ERR_INVALID_ARG_TYPE'
}
```

The only thing I'm changing in my package.json file is 
`"dependencies": {
      "@types/arcgis-js-api": "4.17.0",
  }`
 `"devDependencies": {
      "@arcgis/webpack-plugin": "4.17",
   }`

I'm using node v12.16.0 if that makes any difference.

**Comment by andygup** [comment]
@nbabel thanks. You might try this in your webpack config:

```

 node: {
 process: false,
 global: false,
 Buffer: false,
 setImmediate: false,
 fs: 'empty',
 },

```

**Comment by nbabel** [comment]
Unfortunately, that had no effect.  This is my webpack.config file without those changes:

```
const ArcGISPlugin = require("@arcgis/webpack-plugin");
const LiveReloadPlugin = require('webpack-livereload-plugin');
const options = {
  protocol: 'http',
  port: 35730,
  hostname: 'localhost',
  // appendScriptTag: true,
  delay: 0,

};

module.exports = {
  plugins: [
    new ArcGISPlugin(),
    new LiveReloadPlugin(options)
  ],
  node: {
    process: false,
    global: false,
    fs: "empty"
  }
};
```

**Comment by andygup** [comment]
Gotcha. I have a some questions:

1. Make sure you are running the latest version of the webpack-plugin: `npm install @arcgis/webpack-plugin@latest`. Your version should `4.17.5`
2. Can you paste your `package.json` file here?
3. What is `index-html-webpack-pluginbuffer.js`? It's in the error message you posted. It's sounds like a webpack file but I did a quick search for it and didn't find anything obvious.
4. And `buffer.js` Is that the buffer module from node.js or one of yours? Just double checking.

> 
```js
at Function.from (buffer.js:313:9)
```

5. Can you create a stackblitz app that reproduces this issue?

**Comment by andygup** [comment]
Closing as inactive. Feel free to provide follow-up info, if needed.

**Comment by nbabel** [comment]
Just getting back to this.  It appears using version 4.17.5 has done the trick.  I'm not getting the errors I was before and the map and layers are loading correctly. However, there is a warning in the console:  Failed to create Worker.  Fallback to execute module in main thread.  

It has a message of: Uncaught NetworkError: Failed to execute 'importScripts' on 'WorkerGlobalScope': The script at 'https://js.arcgis.com/4.17.5/dojo/dojo-lite.js' failed to load.  Is dojo-lite.js not included with 4.17.5 anymore?  

Everything appears to be working, so I guess it isn't needed?

**Comment by andygup** [comment]
Good deal. That message is a performance degradation warning because the workers aren't configured correctly.

You might try out @arcgis/core which will go into beta in the week or two: https://www.npmjs.com/package/@arcgis/core. That build uses native ES modules (rather than AMD), won't need to configure webpack and the workers code is embedded.

More info here: https://github.com/Esri/feedback-js-api-next/tree/master/esm-samples/jsapi-angular-cli
